EoL for CM3 plus updates to docs (#4226)
raspberrypi/documentation: documentation/asciidoc/computers/compute-module/introduction.adoc

Unlike **CM4**, CM4S retains the DDR2 SODIMM form factor used in *CM1*, *CM3*, and *CM3+*.

.Compute Module 4S
image::images/cm4s.jpg[alt="Compute Module 4S", width="60%"]

[[cm4]]
=== Compute Module 4

Compute Module 4 (*CM4*) combines the core components of Raspberry Pi 4 with optional flash storage. Key features include:

* *Processor.* Broadcom BCM2711.
* *Memory options.* 1 GB, 2 GB, 4 GB, or 8 GB of RAM.
* *Storage options.* 0 GB (*CM4Lite*), 8 GB, 16 GB, or 32 GB of eMMC memory.
* *Form factor.* Two 100-pin high-density connectors for connecting to the companion carrier board.

Unlike earlier modules (*CM1*, *CM3*, *CM3+*), CM4 moved away from the DDR2 SODIMM form factor to
a dual 100-pin high-density connector layout, which results in a smaller physical footprint. This redesign supports the following additional features:

* Dual HDMI connectors
* PCIe support
* Ethernet connector

.Compute Module 4
image::images/cm4.jpg[alt="Compute Module 4", width="60%"]

[[cm3plus]]
=== Compute Module 3+

Compute Module 3+ (*CM3+*) combines the core components of Raspberry Pi 3 Model B+ with optional flash storage. Key features include:

* *Processor.* Broadcom BCM2837B0.

> * *Memory options*. 1 GB, 2 GB, or 4 GB of RAM.

Reviewer: Appears to have gained memory options. CM3+ only comes in 1GB Ram

Compute Module 3 (*CM3*) combines the core components of Raspberry Pi 3 with an optional 4 GB of flash storage. Key features include:

* *Processor.* Broadcom BCM2837.
* *Memory options.* 1 GB, 2 GB, or 4 GB of RAM.
* *Storage options.* 0 GB (*CM3Lite*) or 4 GB of eMMC flash memory.
* *Form factor.* Standard DDR2 SODIMM module.

.Compute Module 3
image::images/cm3.jpg[alt="Compute Module 3", width="60%"]

[[cm1]]
=== Compute Module 1

Compute Module 1 (*CM1*) combines the core components of Raspberry Pi Model B with an optional 4 GB of flash storage. Key features include:

* *Processor.* Broadcom BCM2835.
* *Memory.* 512 MB of RAM.
* *Storage options.* 0 GB (*CM1Lite*) or 4 GB of eMMC flash memory.
* *Form factor.* Standard DDR2 SODIMM module.

.Compute Module 1
image::images/cm1.jpg[alt="Compute Module 1", width="60%"]

== IO Boards

> A Raspberry Pi Compute Module IO Board is the companion carrier board that provides the necessary connectors to interface with various input/output (I/O) peripherals on your Compute Module. Raspberry Pi Compute Module IO Boards provide the following functionality:

Reviewer 2: Is I/O one that needs to be defined? If so, first use is now line 84, where it is not currently defined.

* Supply power to the Compute Module.
* Connect general-purpose input/output (GPIO) pins to standard pin headers so that you can attach sensors or electronics.
* Expose camera and display interfaces through flat flexible cable (FFC) connectors.
* Route HDMI signals to HDMI connectors for display output.
* Connect USB interfaces to standard USB connectors for peripheral devices.
* Provide LEDs that indicate power and activity status.
* Enable eMMC programming over USB for flashing the module's onboard storage.
* Expose PCIe lanes through connectors so that you can attach storage or peripheral devices like SSDs or network adapters.

Raspberry Pi IO Boards are general-purpose boards designed for development, testing, and prototyping Compute Modules. For production use, you might design a smaller, custom carrier board that includes only the connectors you need for your use case.

> === IO Boards and compatibility

Reviewer 2: ```suggestion
[[io-board-compatibility]]
=== IO Boards and compatibility
```

This should fix the build break.

Not all IO Boards work with all Compute Module models. The following table summarises Raspberry Pi Compute Module IO Boards in reverse chronological order, listing their compatible Compute Modules (which include Lite versions), power input, and size. For more information about each of these boards, including available interfaces, see the following dedicated sections on this page.

[cols="1,1,1,1", options="header"]
|===
|IO Board|Compatible CM|Power input|Size

| <<cm5io, *Compute Module 5 IO Board (CM5IO)*>> (2024)
| <<cm5, CM5>>; CM4 with reduced functionality
| 5 V USB Type-C external supply
|160 mm × 90 mm

| <<cm4io, *Compute Module 4 IO Board (CM4IO)*>> (2020)
| <<cm4, CM4>>; CM5 with reduced functionality
| 12 V or 5 V through the DC barrel jack
|160 mm × 90 mm

| <<cmio, *Compute Module IO Board version 3 (CMIO3)*>> (2017)
| <<cm1, CM1>>, <<cm3, CM3>>, <<cm3plus, CM3+>>, and <<cm4S, CM4S>>
| 12 V or 5 V through the barrel jack
| 160 mm × 100 mm

| <<cmio, *Compute Module IO Board version 1 (CMIO; CMIO1)*>> (2014)
| <<cm1, CM1>>

> | 5 V through the barrel jack

Reviewer: No it isn't is micro USB

.Compute Module 5 IO Board
image::images/cm5io.png[alt="Compute Module 5 IO Board", width="60%"]

The Compute Module 5 IO Board (CM5IO) provides the following:

* *Power and control connectors.*
** USB-C power using the same standard as Raspberry Pi 5: 5 V at 5 A (25 W) or 5 V at 3 A (15 W) with a 600 mA peripheral limit.
** A power button for CM5.
** Real-time clock (RTC) battery socket.
* *Video and display connectors.*
** Two HDMI connectors.
** Two MIPI DSI/CSI-2 combined display/camera FPC connectors (22-pin, 0.5 mm pitch cable).
* *Networking and connectivity connectors.*
** Two USB 3.0 (Type-A) connectors for keyboards, storage, or peripherals.
** A USB 2.0 (Type-C) connector for flashing CM5 or additional peripherals.
** A Gigabit Ethernet RJ45 with PoE support.
* *Expansion and storage options.*
** A M.2 M key PCIe socket compatible with the 2230, 2242, 2260, and 2280 form factors.
** A microSD card slot (only for use with *CM5Lite*, which has no eMMC; other variants ignore the slot).
** HAT footprint with 40-pin GPIO connector.
* *Configuration options.*
** Jumpers to disable features such as eMMC boot, EEPROM write, and wireless connectivity.

> ** Selectable R4/R5 Vref voltage to support different voltage requirements.

Reviewer: ** Selectable 1.8 V  or  3.3 V GPIO Voltage selection

The Compute Module 4 IO Board (CM4IO) provides the following:

* *Power and control connectors.*
** 12 V input through barrel jack; supports up to 26 V if PCI is unused.
** Real-time clock (RTC) battery socket.
* *Video and display connectors.*
** Two HDMI connectors.
** Two MIPI DSI display FPC connectors (22-pin, 0.5 mm pitch cable).
** Two MIPI CSI-2 camera FPC connectors (22-pin, 0.5 mm pitch cable).
* *Networking and connectivity connectors.* Two USB 2.0 connectors.
** A micro USB upstream port.
** A Gigabit Ethernet RJ45 with PoE support.
* *Expansion and storage options.*
** PCIe Gen 2 socket.
** A microSD card slot (only for use with *CM4Lite*, which has no eMMC; other variants ignore the slot).
** HAT footprint with 40-pin GPIO connector and PoE header
* *Configuration options.*
** Jumpers to disable features such as eMMC boot, EEPROM write, and wireless connectivity.
** Selectable R4/R5 Vref voltage to support different voltage requirements.
* *Fan connector.* Fan connector supporting standard 12 V fans with PWM drive.

[[cmio]]
=== Compute Module IO Board (versions 1 and 3)

.Compute Module IO Board
image::images/cmio.jpg[alt="Compute Module IO Board", width="60%"]

> There are two vairants of the Compute Module IO Board:

Reviewer 2: ```suggestion
There are two variants of the Compute Module IO Board:
```

* Version 1 (CMIO), compatible only with <<cm1, CM1>>.
* Version 3 (CMIO3), compatible with <<cm1, CM1>>, <<cm3, CM3>>, <<cm3plus, CM3+>>, and <<cm4s, CM4S>>. This version adds a microSD card slot that doesn't exist on CMIO (version 1).

The Compute Module IO Board (CMIO and CMIO3) provides the following:

* *Power and control connectors.* 5 V input through GPIO or external PSU.
* *Video and display connectors.*

> ** One HDMI 1.3a connector.

Reviewer: ** One Full size Type A HDMI supporting HDMI 1.3a.

Reviewer: okay now I've read the context a little more lets remove " supporting HDMI 1.3a" as with the CM4S its HDMI 1.4

. *Connect or remove the fan.* 
** If using the pre-installed fan, plug the fan into the four-pin fan connector labelled *FAN (J14)* on your CM5IO. Ensure that the fan is properly seated and connected to the power header on your CM5IO.
** If you want to remove the fan, unscrew the four corner screws of the fan from the underside of the top of the case.
. *Optionally, attach an external antenna.* If you want to install an antenna, follow the instructions in <<antenna, Antenna (CM4 and CM5)>> on this page.
. *Optionally, attach the cooler.* If you want to install a cooler, follow the instructions in <<cooler, CM5 Cooler>>. If you're also attaching an antenna, attach the antenna's U.FL connector first for easier access.
. *Optionally, attach a camera or display.* If you're using a camera or a display, pass the flat cable through one of the slots at the back of the case and connect it to one of the *CAM/DISP* ports on your CM5IO.
. *Optionally, mount additional accessories.*
** To install an M.2 SSD, insert it into the M.2 slot in the bottom-right corner of the CM5IO and secure it on the opposite end with mounting screw.

> ** To install a PoE+ HAT, align it with the 40-pin GPIO header and the mounting posts such that the hat covers the battery slot, then press it firmly into place and secure it with screws.

Reviewer 2: ```suggestion
** To install a PoE+ HAT, align it with the 40-pin GPIO header and the mounting posts such that the HAT covers the battery slot, then press it firmly into place and secure it with screws.
```

> . *Close the case.* Fold the top of the case back onto the base of the case, aligning the screw holes on the left and right sides of the case, and the power button on the back of the case. Screw the four screws back into place using a cross-head screwdriver, taking care not to overtighten them.

Reviewer: cross head screw driver isn't descriptive enough. I assume you mean "Philips screwdriver"

Author: I was advised to use "cross-head screws" instead of being too specific for the Touch Display stuff, but I will change it to Phillips Screwdriver if that's what you think it more appropriate in this case.

. *Fasten the SMA connector into place.* Twist the retaining hexagonal nut and washer onto the SMA connector in a clockwise direction until it sits securely in place. Avoid excessive twisting when tightening to prevent damage.
. *Attach the SMA connector to the antenna.* Insert the SMA connector into the antenna port with the antenna facing outward and twist the antenna clockwise to secure it. 
. *Adjust the antenna.* Move the antenna into its final position by turning it up to a 90 degree angle.

.CM4 and CM5 antenna assembly diagram
image::images/cm4-cm5-antenna-assembly.svg[alt="CM4 and CM5 antenna assembly diagram", width="60%"]

To use the Antenna with your Compute Module, add a `dtoverlay` instruction in xref:../computers/config_txt.adoc[`/boot/firmware/config.txt`]. Add the following line to the end of the `config.txt` file: `dtparam=ant2`

[[cooler]]
=== CM5 Cooler

The CM5 Cooler is a passive heat sink that helps dissipate heat from your CM5, improving CPU performance and longevity. 

.CM5 cooler
image::images/cm5-cooler.jpg[alt="CM5 Cooler", width="60%"]

> The CM5 Cooler dimensions are 41 mm × 56 mm × 12.7 mm. The cooler has an aluminum profile with an underside that's made of thermally conductive silicone. Newer versions of the <<case, CM5IO Case>> fit both the cooler and pre-installed fan inside. If you have an older version of the CM5IO Case, you must remove the fan from the case to allow space for the cooler.

Reviewer 2: ```suggestion
The CM5 Cooler dimensions are 41 mm × 56 mm × 12.7 mm. The cooler has an aluminium profile with an underside that's made of thermally conductive silicone. Newer versions of the <<case, CM5IO Case>> fit both the cooler and pre-installed fan inside. If you have an older version of the CM5IO Case, you must remove the fan from the case to allow space for the cooler.
```

> The CM5 Cooler dimensions are 41 mm × 56 mm × 12.7 mm. The cooler has an aluminum profile with an underside that's made of thermally conductive silicone. Newer versions of the <<case, CM5IO Case>> fit both the cooler and pre-installed fan inside. If you have an older version of the CM5IO Case, you must remove the fan from the case to allow space for the cooler.

Reviewer: The CM5 Cooler dimensions are 41 mm × 56 mm × 12.7 mm. The cooler is an aluminum heatsink with a conductive silicone pad on the bottom. Newer versions of the <<case, CM5IO Case>> enable both cooler and the pre-installed fan to be used at the same time. If you have an older version of the CM5IO Case, you must remove the fan from the case to allow space for the cooler.

.CM5 cooler physical specification
image::images/cm5-cooler-physical.png[alt="CM5 Cooler physical specification", width="80%"]

> To mount the cooler to your CM5, attach the thermally conductive silicone at the bottom of the cooler to the top of your CM5. Align the cut-out in the heatsink with the antenna https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector]. Optionally, fasten screws in the mounting points found in each corner to secure the cooler. If you omit the screws, the bond between your cooler and your CM5 improves through time and use.

Reviewer: To mount the cooler to your CM5, remove the protective paper form the silicon pad on the bottom of the heatsink. Align the cut-out in the heatsink with the antenna , the triangular shaped area close to the edge of the CM5. Optionally, the cooler maybe screwed down using the mounting points in each corner. If you omit the screws, the bond between your cooler and your CM5 improves through time and use.

Reviewer: Yeah not the greatest description , not sure what else to say.

[cols="1,1,1,1,1,1", options="header"]
|===
|Model|Based on|SoC|CPU|GPU|Form factor

| <<cm5, *CM5*>> (2024)
| Raspberry Pi 5
| Broadcom BCM2712
|VideoCore VII
| 4 × Cortex-A76 at 2.4 GHz
|Dual 100-pin connectors

| <<cm4s, *CM4S*>> (2022)
| Raspberry Pi 4 Model B (in CM3 form factor)
| Broadcom BCM2711
|VideoCore VI
| 4 × Cortex-A72 at 1.5 GHz
|DDR2 SODIMM

| <<cm4, *CM4*>> (2020)
| Raspberry Pi 4 Model B
| Broadcom BCM2711
|VideoCore VI
| 4 × Cortex-A72 at 1.5 GHz
|Dual 100-pin connectors

| <<cm3plus, *CM3+*>> (2019)
| Raspberry Pi 3 Model B+
| Broadcom BCM2837B0
|VideoCore IV
| 4 × Cortex-A53 at 1.2 GHz
|DDR2 SODIMM

| <<cm3, *CM3*>> (2017; discontinued October 2025)
| Raspberry Pi 3 Model B
| Broadcom BCM2837
|VideoCore IV
| 4 × Cortex-A53 at 1.2 GHz
|DDR2 SODIMM

| <<cm1, *CM1*>>(2014)
| Raspberry Pi Model B
| Broadcom BCM2835
|VideoCore IV
| 1 × ARM1176JZF-S at 700 MHz
|DDR2 SODIMM

|===

> NOTE: Compute Module 2 (CM2) isn't included in this documentation because it was never released publicly. Compute Module 1 (CM1) was officially followed by Compute Module 3 (CM3).

Author: ```suggestion
NOTE: Compute Module 2 (CM2) was never released publicly. Compute Module 1 (CM1) was officially followed by Compute Module 3 (CM3).
```

* *Processor.* Broadcom BCM2711.
* *Memory options.* 1 GB, 2 GB, 4 GB, or 8 GB of RAM.
* *Storage options.* 0 GB (*CM4Lite*), 8 GB, 16 GB, or 32 GB of eMMC memory.
* *Form factor.* Two 100-pin high-density connectors for connecting to the companion carrier board.

Unlike earlier modules (*CM1*, *CM3*, *CM3+*), CM4 moved away from the DDR2 SODIMM form factor to
a dual 100-pin high-density connector layout, which results in a smaller physical footprint. This redesign supports the following additional features:

* Dual HDMI connectors
* PCIe support
* Ethernet connector

.Compute Module 4
image::images/cm4.jpg[alt="Compute Module 4", width="60%"]

[[cm3plus]]
=== Compute Module 3+

Compute Module 3+ (*CM3+*) combines the core components of Raspberry Pi 3 Model B+ with optional flash storage. Key features include:

* *Processor.* Broadcom BCM2837B0.
* *Memory*. 1 GB of RAM.
* *Storage options.* 0 GB (*CM3+Lite*) or 8 GB, 16 GB, or 32 GB of eMMC flash memory.
* *Form factor.* Standard DDR2 SODIMM module.

.Compute Module 3+
image::images/cm3-plus.jpg[alt="Compute Module 3+", width="60%"]

[[cm3]]
=== Compute Module 3

> IMPORTANT: Raspberry Pi Compute Module 3 (CM3) and Compute Module 3 Lite (CM3Lite) have reached End-of-Life (EoL) due to the discontinuation of the core SoC used in these products. The official EoL date was 16 October, 2025. The closest equivalent to CM3 is Raspberry Pi Compute Module 3+ (CM3+), which offers the same mechanical footprint, improved thermal design, and a BCM2837B0 processor. For new designs requiring the SODIMM form factor, we recommend <<cm3plus, Compute Module 3+>> or <<cm4s, Compute Module 4S>>; for other designs, we recommend <<cm4, Compute Module 4>> or <<cm5, Compute Module 5>>. For more information, see the official https://pip-assets.raspberrypi.com/categories/1284-eol-notices/documents/RP-009286-PC-1-OBS2_%20Raspberry%20Pi%20Compute%20Module%203%20and%20Raspberry%20Pi%20Compute%20Module%203%20Lite.pdf?disposition=inline[Obsolescence Notice].

Author: ```suggestion
IMPORTANT: Raspberry Pi Compute Module 3 (CM3) and Compute Module 3 Lite (CM3Lite) have reached End-of-Life (EoL) due to the discontinuation of the core SoC used in these products. The official EoL date was 16 October, 2025. The closest equivalent to CM3 is Raspberry Pi <<cm3plus, Compute Module 3+ (CM3+)>>, which offers the same mechanical footprint, improved thermal design, and a BCM2837B0 processor, and so is recommended for existing designs. For new designs requiring the SODIMM form factor, we recommend <<cm4s, Compute Module 4S>>. For all other new designs, we recommend <<cm4, Compute Module 4>> or <<cm5, Compute Module 5>>. For more information, see the official https://pip-assets.raspberrypi.com/categories/1284-eol-notices/documents/RP-009286-PC-1-OBS2_%20Raspberry%20Pi%20Compute%20Module%203%20and%20Raspberry%20Pi%20Compute%20Module%203%20Lite.pdf?disposition=inline[Obsolescence Notice].```

> IMPORTANT: Raspberry Pi Compute Module 3 (CM3) and Compute Module 3 Lite (CM3Lite) have reached End-of-Life (EoL) due to the discontinuation of the core SoC used in these products. The official EoL date was 16 October, 2025. The closest equivalent to CM3 is Raspberry Pi <<cm3plus, Compute Module 3+ (CM3+)>>, which offers the same mechanical footprint, improved thermal design, and a BCM2837B0 processor, and so is recommended for existing designs. For new designs requiring the SODIMM form factor, we recommend <<cm4s, Compute Module 4S>>. For all other new designs, we recommend <<cm4, Compute Module 4>> or <<cm5, Compute Module 5>>. For more information, see the official https://pip-assets.raspberrypi.com/categories/1284-eol-notices/documents/RP-009286-PC-1-OBS2_%20Raspberry%20Pi%20Compute%20Module%203%20and%20Raspberry%20Pi%20Compute%20Module%203%20Lite.pdf?disposition=inline[Obsolescence Notice].```

Author: ```suggestion
IMPORTANT: Raspberry Pi Compute Module 3 (CM3) and Compute Module 3 Lite (CM3Lite) have reached End-of-Life (EoL) due to the discontinuation of the core SoC used in these products. The official EoL date was 16 October, 2025. The closest equivalent to CM3 is Raspberry Pi <<cm3plus, Compute Module 3+>>, which offers the same mechanical footprint, improved thermal design, and a BCM2837B0 processor, and so is recommended for existing designs. For new designs requiring the SODIMM form factor, we recommend <<cm4s, Compute Module 4S>>. For all other new designs, we recommend <<cm4, Compute Module 4>> or <<cm5, Compute Module 5>>. For more information, see the official https://pip-assets.raspberrypi.com/categories/1284-eol-notices/documents/RP-009286-PC-1-OBS2_%20Raspberry%20Pi%20Compute%20Module%203%20and%20Raspberry%20Pi%20Compute%20Module%203%20Lite.pdf?disposition=inline[Obsolescence Notice].```
```

[[cmio]]
=== Compute Module IO Board (versions 1 and 3)

.Compute Module IO Board
image::images/cmio.jpg[alt="Compute Module IO Board", width="60%"]

There are two variants of the Compute Module IO Board:

* Version 1 (CMIO), compatible only with <<cm1, CM1>>.
* Version 3 (CMIO3), compatible with <<cm1, CM1>>, <<cm3, CM3>>, <<cm3plus, CM3+>>, and <<cm4s, CM4S>>. This version adds a microSD card slot that doesn't exist on CMIO (version 1).

The Compute Module IO Board (CMIO and CMIO3) provides the following:

* *Power and control connectors.* 5 V input through GPIO or a micro USB connector.
* *Video and display connectors.*
** One Full size Type A HDMI.
** Two MIPI DSI display FPC connectors (22-pin, 0.5 mm pitch cable).
** Two MIPI CSI-2 camera FPC connectors (22-pin, 0.5 mm pitch cable).
** One 24-bit DPI (parallel RGB) interface.
* *Networking and connectivity connectors.* One USB 2.0 Type-A connector.
* *Expansion and storage options.*
** 200-pin DDR2 SODIMM connector.
** 46 GPIO pins.
** (CMIO3 only) A microSD card slot (only for use with *CM3Lite*, which has no eMMC).

== CM5 and CM5IO accessories

Raspberry Pi offers the following accessories for CM5 and CM5IO:

> * <<case, CM5IO Case>>, a two-piece metal enclosure for a CM5I0 (and attached CM5). The case also optionally fits an antenna and cooler.

Author: ```suggestion
* <<case, CM5IO Case>>, a two-piece metal enclosure for a CM5IO (and attached CM5). The case also optionally fits an antenna and cooler.
```

image::images/cm4-cm5-antenna-assembly.svg[alt="CM4 and CM5 antenna assembly diagram", width="60%"]

To use the Antenna with your Compute Module, add a `dtoverlay` instruction in xref:../computers/config_txt.adoc[`/boot/firmware/config.txt`]. Add the following line to the end of the `config.txt` file: `dtparam=ant2`

[[cooler]]
=== CM5 Cooler

The CM5 Cooler is a passive heat sink that helps dissipate heat from your CM5, improving CPU performance and longevity. 

.CM5 cooler
image::images/cm5-cooler.jpg[alt="CM5 Cooler", width="60%"]

The CM5 Cooler dimensions are 41 mm × 56 mm × 12.7 mm. The cooler is an aluminum heat sink with a conductive silicone pad on the bottom. Newer versions of the <<case, CM5IO Case>> allow both the cooler and pre-installed fan to be used inside the case at the same time. If you have an older version of the CM5IO Case, you must remove the fan from the case to allow space for the cooler.

.CM5 cooler physical specification
image::images/cm5-cooler-physical.png[alt="CM5 Cooler physical specification", width="80%"]

To mount the cooler to your CM5:

. Remove the protective paper from the silicone pad on the bottom of cooler.

> . Attach the silicone at the bottom of the cooler to the top of your CM5. Align the cut-out in the heat sink with the antenna https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector].

Author: ```suggestion
. Attach the silicone at the bottom of the cooler to the top of your CM5. Align the cut-out in the heat sink with the antenna https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] and the on board antenna. Ensure that both the U.FL connector on the Compute Module, and the trapezoid shape area next to the U.FL connector, are visible through the cutout.
```

> . Attach the silicone at the bottom of the cooler to the top of your CM5. Align the cutout in the cooler with the https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] and the on-board antenna (the trapezoid shaped area on the board next to the U.FL connector), ensuring both are visible through the cutout.

Author: ```suggestion
. Attach the silicone at the bottom of the cooler to the top of your CM5. Place the cooler on your CM5 such that the cutout in the cooler is above  the on-board antenna (the trapezoid-shaped area on the left of a CM5) and the https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] next to it (if it has one).
```

> . Attach the silicone at the bottom of the cooler to the top of your CM5. Align the cut-out in the heat sink with the antenna https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] and the on board antenna. Ensure that both the U.FL connector on the Compute Module, and the trapezoid shape area next to the U.FL connector, are visible through the cutout.

Author: ```suggestion
. Attach the silicone at the bottom of the cooler to the top of your CM5. Align the cut-out in the heat sink with the antenna https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] and the on board antenna. Ensure that both the U.FL connector on the Compute Module, and the trapezoid shape area next to the U.FL connector (the on-board antenna), are visible through the cutout.
```

> . Attach the silicone at the bottom of the cooler to the top of your CM5. Align the cut-out in the heat sink with the antenna https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] and the on board antenna. Ensure that both the U.FL connector on the Compute Module, and the trapezoid shape area next to the U.FL connector (the on-board antenna), are visible through the cutout.

Author: ```suggestion
. Attach the silicone at the bottom of the cooler to the top of your CM5. Align the cut-out in the heat sink with the antenna https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] and the on-board antenna. Ensure that both the U.FL connector on the Compute Module, and the trapezoid shape area next to the U.FL connector (the on-board antenna), are visible through the cutout.
```

> . Attach the silicone at the bottom of the cooler to the top of your CM5. Align the cut-out in the heat sink with the antenna https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] and the on-board antenna. Ensure that both the U.FL connector on the Compute Module, and the trapezoid shape area next to the U.FL connector (the on-board antenna), are visible through the cutout.

Author: ```suggestion
. Attach the silicone at the bottom of the cooler to the top of your CM5. Align the cutout in the cooler with the https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] and the on-board antenna (the trapezoid shaped area on the board next to the U.FL connector), ensuring both are visible through the cutout.
```

This page:

* Summarises the available Raspberry Pi Compute Module and IO Board models, including information about their compatibility and key features.
* Describes the accessories available for Compute Module 5 (CM5) and its IO Board (CM5IO).
* Explains how to flash and boot Raspberry Pi Compute Modules.
* Explains how to configure the EEPROM bootloader of a Compute Module.
* Explains how to wire and enable peripherals like cameras and displays using Device Tree and overlays.
* Provides links to datasheets, schematics, and design resources.

== Compute Modules

Raspberry Pi Compute Modules are *system-on-module (SoM)* variants of the flagship Raspberry Pi single-board computers (SBC). They're designed for industrial and commercial applications, such as digital signage, thin clients, and process automation. Many developers and system designers choose Compute Modules over flagship Raspberry Pi models for their compact design, flexibility, and support for on-board eMMC storage.

> Raspberry Pi Compute Modules are available in several variants, differing in memory and embedded Multi-Media Card (eMMC) flash storage capacity (soldered onto the board). Like SD cards, eMMC provides persistent storage with low power consumption. Unlike SD cards, eMMC behaves like an internal disk and includes built-in features that improve reliability. *Lite* models have no on-board storage, and are sometimes referred to with the suffix *Lite*, for example, "CM5Lite".

Reviewer 3: Perhaps?
```suggestion
Raspberry Pi Compute Modules are available in several variants, differing in memory and embedded Multi-Media Card (eMMC) flash storage capacity (soldered onto the board). Like SD cards, eMMC provides persistent storage with low power consumption. Unlike SD cards, eMMC behaves like an internal disk and includes built-in features that improve reliability. *Lite* models have no on-board eMMC storage, and are sometimes referred to with the suffix *Lite*, for example, "CM5Lite".
```

Also (just to complicate things) there are variants of the Compute Modules available with or without WiFi.

Author: This is mostly what's already included in the current docs (second paragraph), except that instead of "with minimal energy impact", I tried to be more specific by saying "low power consumption". Is there anything in particular you'd like me to say about Wi-Fi?

Compute Module 3 (*CM3*) combines the core components of Raspberry Pi 3 with an optional 4 GB of flash storage. Key features include:

* *Processor.* Broadcom BCM2837.
* *Memory.* 1 GB of RAM.
* *Storage options.* 0 GB (*CM3Lite*) or 4 GB of eMMC flash memory.
* *Form factor.* Standard DDR2 SODIMM module.

.Compute Module 3
image::images/cm3.jpg[alt="Compute Module 3", width="60%"]

[[cm1]]
=== Compute Module 1

Compute Module 1 (*CM1*) combines the core components of Raspberry Pi Model B with an optional 4 GB of flash storage. Key features include:

* *Processor.* Broadcom BCM2835.
* *Memory.* 512 MB of RAM.

> * *Storage options.* 0 GB (*CM1Lite*) or 4 GB of eMMC flash memory.

Reviewer 3: I don't think there was a CM1Lite, it was the CM3 that first introduced the "Lite" variant?
( https://www.raspberrypi.com/products/compute-module-1/ seems to back up my recollection )

Author: Oh, I though all "0 GB" versions were called "Lite"? Or are you saying it only came with 4 GB of eMMC flash memory?

* On CM4IO and CM5IO, expose PCIe through connectors so that you can attach storage or peripheral devices like SSDs or network adapters.

Raspberry Pi IO Boards are general-purpose boards designed for development, testing, and prototyping Compute Modules. For production use, you might design a smaller, custom carrier board that includes only the connectors you need for your use case.

[[io-board-compatibility]]
=== IO Boards and compatibility

Not all IO Boards work with all Compute Module models. The following table summarises Raspberry Pi Compute Module IO Boards in reverse chronological order, listing their compatible Compute Modules (which include Lite versions), power input, and size. For more information about each of these boards, including available interfaces, see the following dedicated sections on this page.

[cols="1,1,1,1", options="header"]
|===
|IO Board|Compatible CM|Power input|Size

| <<cm5io, *Compute Module 5 IO Board (CM5IO)*>> (2024)
| <<cm5, CM5>>; CM4 with reduced functionality
| 5 V USB Type-C external supply
|160 mm × 90 mm

| <<cm4io, *Compute Module 4 IO Board (CM4IO)*>> (2020)
| <<cm4, CM4>>; CM5 with reduced functionality

> | 12 V or 5 V through the DC barrel jack

Reviewer: 5v through he GPIO header or 12v through the DC jack.  

FYI Can't do power via the micro USB

** HAT footprint with 40-pin GPIO connector and PoE header
* *Configuration options.*
** Jumpers to disable features such as eMMC boot, EEPROM write, and wireless connectivity.
** Selectable 1.8 V or 3.3 V GPIO voltage.
* *Fan connector.* Fan connector supporting standard 12 V fans with PWM drive.

[[cmio]]
=== Compute Module IO Board (versions 1 and 3)

.Compute Module IO Board
image::images/cmio.jpg[alt="Compute Module IO Board", width="60%"]

There are two variants of the Compute Module IO Board:

* Version 1 (CMIO), compatible only with <<cm1, CM1>>.
* Version 3 (CMIO3), compatible with <<cm1, CM1>>, <<cm3, CM3>>, <<cm3plus, CM3+>>, and <<cm4s, CM4S>>. This version adds a microSD card slot that doesn't exist on CMIO (version 1).

The Compute Module IO Board (CMIO and CMIO3) provides the following:

* *Power and control connectors.* 5 V input through GPIO or a micro USB connector.
* *Video and display connectors.*
** One Full size Type A HDMI.
** Two MIPI DSI display FPC connectors (22-pin, 0.5 mm pitch cable).
** Two MIPI CSI-2 camera FPC connectors (22-pin, 0.5 mm pitch cable).

> ** One 24-bit DPI (parallel RGB) interface.

Reviewer 3: I think that this is just one of the modes of the GPIO pins, so I'm not sure if it's worth listing this separately as a CMIO feature?

Reviewer: CVBS ( TVOUT ) is also available on the Pin header too. DPI is an alternate mode of some of the GPIO pins. Other IO board don't have DPI in the list yet support DPI .

Author: So, remove it?

* *Networking and connectivity connectors.* One USB 2.0 Type-A connector.
* *Expansion and storage options.*
** 200-pin DDR2 SODIMM connector.
** 46 GPIO pins.
** (CMIO3 only) A microSD card slot (only for use with *CM3Lite*, which has no eMMC).

== CM5 and CM5IO accessories

Raspberry Pi offers the following accessories for CM5 and CM5IO:

* <<case, CM5IO Case>>, a two-piece metal enclosure for a CM5IO (and attached CM5). The case also optionally fits an antenna and cooler.

> * <<antenna, Antenna (CM4 and CM5)>>, a 2.4 GHz and 5 GHz antenna to for wireless connectivity through CM5. The Raspberry Pi Antenna is also compatible with CM4.

Reviewer 3: ```suggestion
* <<antenna, Antenna (CM4 and CM5)>>, a 2.4 GHz and 5 GHz antenna for wireless connectivity through CM5. The Raspberry Pi Antenna is also compatible with CM4.
```

. *Connect or remove the fan.* 
** If using the pre-installed fan, plug the fan connector into the four-pin fan socket labelled *FAN (J14)* on your CM5IO.
** If you want to remove the fan, unscrew the four corner screws of the fan from the underside of the top of the case.
. *Optionally, attach an external antenna.* If you want to install an antenna, follow the instructions in <<antenna, Antenna (CM4 and CM5)>> on this page.
. *Optionally, attach the cooler.* If you want to install a cooler, follow the instructions in <<cooler, CM5 Cooler>>. If you're also attaching an antenna, attach the antenna's U.FL connector first for easier access.
. *Optionally, attach a camera or display.* If you're using a camera or a display, pass the flat cable through one of the slots at the back of the case and connect it to one of the *CAM/DISP* ports on your CM5IO.
. *Optionally, install an M.2 SSD.* If you want to install an M.2 SSD, insert it into the M.2 slot in the bottom-right corner of the CM5IO and secure it on the opposite end with a mounting screw.

> . *Optionally, install a HAT.* If you want to install a HAT, align it with the 40-pin GPIO header and the mounting posts such that the HAT covers the battery slot, then press it firmly into place and secure it with screws.

Reviewer 3: Optionally connect an RTC battery? :slightly_smiling_face:

Author: Is that relevant to the whether things fit inside the CM5IO Case?

Author: No, I found a couple of sets of instructions for the CM5IO Case and they weren't in line with each other, so I tried to pull both sets of instructions together. I also got my hands on a CM5IO Case, antenna, and cooler, and then asked about how the M.2 SSD and HAT were attached, which were originally more casually referred to as "accessories connected to the IO Board, such as an M.2 SSD or a PoE+ HAT+" in the Product Brief. Originally, I kept a similar level of generality, but then thought it might be useful to explicitly state where they would be placed. To keep in line with the Product Brief as much as possible, I originally kept the specificity of the "PoE+ HAT+" but this was corrected the more generic "HAT".

Author: If you and/or @dp111 think it would be useful to include instructions for connecting an RTC battery, I can do that tomorrow.

. *Close the case.* Fold the top of the case back onto the base of the case, aligning the screw holes on the left and right sides of the case, and the power button on the back of the case. Screw the four screws back into place using a Phillips screwdriver, taking care not to overtighten them.

NOTE: The SD card slot is a push-push slot. To insert an SD card, push it into the SD card slot with the contacts facing downwards. To remove it, push it inwards towards the slot to release it and then pull it out.

[[antenna]]
=== Antenna (CM4 and CM5)

The Raspberry Pi Antenna Kit provides a certified external antenna to boost wireless reception on a CM4 or CM5.

.CM4 and CM5 antenna
image::images/cm4-cm5-antenna.jpg[alt="The Antenna, connected to CM4", width="60%"]

> The antenna supports dual-band Wi-Fi and attaches to the https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] on your CM4 or CM5. The antenna is 108 mm at full height; the SMA to U.FL cable is 205 mm.

Reviewer 3: ```suggestion
The antenna supports dual-band Wi-Fi and attaches to the https://en.wikipedia.org/wiki/Hirose_U.FL[U.FL connector] on your CM4 or CM5. The antenna is 108 mm at full height; the SMA to U.FL cable is 205 mm long.
```
?

.CM4 and CM5 antenna physical specification
image::images/cm4-cm5-antenna-physical.png[alt="CM4 and CM5 antenna physical specification", width="80%"]

You can use the antenna with the <<case, CM5IO Case>>. To attach the antenna to your Compute Module through the CM5IO Case, complete the first four steps outlined in <<case, CM5IO Case>>, and then complete following steps:

> . *Connect the U.FL connector.* Connect the U.FL connector on the antenna cable to the U.FL-compatible connector on your Compute Module, next to the top-left mounting hole of your CM5. Do this before attaching a cooler (if using one) because the cooler can make it harder to attach the U.FL connector.

Reviewer 3: I wonder if it might be better to connect the U.FL connector to the CM5 _after_ attaching the SMA connector to the case, given that the U.FL connector seems more fragile? :thinking:

Author: The order of steps is to be in line with the numbered diagram

. *Insert the SMA connector.* Remove the rubber plug from the antenna port on the inside of the CM5IO Case. Then, from the inside of the case, push the SMA connector with the (flat side up) into the antenna port so that it extends through and is accessible from the outside.
. *Fasten the SMA connector into place.* Twist the retaining hexagonal nut and washer onto the SMA connector in a clockwise direction until it sits securely in place. Avoid excessive twisting when tightening to prevent damage.

> . *Attach the SMA connector to the antenna.* Insert the SMA connector into the antenna port with the antenna facing outward and twist the antenna clockwise to secure it. 

Reviewer 3: Perhaps this should say "Attach the antenna to the SMA connector" ?

. *Fasten the SMA connector into place.* Twist the retaining hexagonal nut and washer onto the SMA connector in a clockwise direction until it sits securely in place. Avoid excessive twisting when tightening to prevent damage.
. *Attach the antenna to the SMA connector.* Insert the SMA connector into the antenna port with the antenna facing outward and twist the antenna clockwise to secure it. 
. *Adjust the antenna.* Move the antenna into its final position by turning it up to a 90 degree angle.

You can now complete the remaining steps outlined in <<case, CM5IO Case>> for mounting a CM5IO inside your case.

.CM4 and CM5 antenna assembly diagram
image::images/cm4-cm5-antenna-assembly.svg[alt="CM4 and CM5 antenna assembly diagram", width="60%"]

To use the Antenna with your Compute Module, add a `dtparam` instruction in xref:../computers/config_txt.adoc[`/boot/firmware/config.txt`]. Add the following line to the end of the `config.txt` file: `dtparam=ant2`

[[cooler]]
=== CM5 Cooler

The CM5 Cooler is a passive heat sink that helps dissipate heat from your CM5, improving CPU performance and longevity. 

.CM5 cooler
image::images/cm5-cooler.jpg[alt="CM5 Cooler", width="60%"]

> The CM5 Cooler dimensions are 41 mm × 56 mm × 12.7 mm. The cooler is an aluminum heat sink with a conductive silicone pad on the bottom. Newer versions of the <<case, CM5IO Case>> allow both the cooler and pre-installed fan to be used inside the case at the same time. If you have an older version of the CM5IO Case, you must remove the fan from the case to allow space for the cooler.

Reviewer 3: ```suggestion
The CM5 Cooler dimensions are 41 mm × 56 mm × 12.7 mm. The cooler is an aluminium heat sink with a conductive silicone pad on the bottom. Newer versions of the <<case, CM5IO Case>> allow both the cooler and pre-installed fan to be used inside the case at the same time. If you have an older version of the CM5IO Case, you must remove the fan from the case to allow space for the cooler.
```

image::images/cmio.jpg[alt="Compute Module IO Board", width="60%"]

There are two variants of the Compute Module IO Board:

* Version 1 (CMIO), compatible only with <<cm1, CM1>>.
* Version 3 (CMIO3), compatible with <<cm1, CM1>>, <<cm3, CM3>>, <<cm3plus, CM3+>>, and <<cm4s, CM4S>>. This version adds a microSD card slot that doesn't exist on CMIO (version 1).

The Compute Module IO Board (CMIO and CMIO3) provides the following:

* *Power and control connectors.* 5 V input through GPIO or a micro USB connector.
* *Video and display connectors.*
** One Full size Type A HDMI.
** Two MIPI DSI display FPC connectors (22-pin, 0.5 mm pitch cable).
** Two MIPI CSI-2 camera FPC connectors (22-pin, 0.5 mm pitch cable).
* *Networking and connectivity connectors.* One USB 2.0 Type-A connector.
* *Expansion and storage options.*
** 200-pin DDR2 SODIMM connector.
** 46 GPIO pins.
** (CMIO3 only) A microSD card slot (only for use with *CM3Lite*, which has no eMMC).

== CM5 and CM5IO accessories

Raspberry Pi offers the following accessories for CM5 and CM5IO:

* <<case, CM5IO Case>>, a two-piece metal enclosure for a CM5IO (and attached CM5). The case also optionally fits an antenna and cooler.

> * <<antenna, Antenna (CM4 and CM5)>>, a 2.4 GHz and 5 GHz antenna for wireless connectivity through CM5. The Raspberry Pi Antenna is also compatible with CM4.

Author: In the future, it might be useful to have a more fleshed out "accessories" section that isn't limited to CM5
